gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -659,13 +733,13 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 		topOfRecycledChunk = ((uint8_t *)addrTop) + recycleEntrySize;
 		/* Recycle the remaining entry back onto the free list (if applicable) */
 		if (recycleHeapChunk(addrTop, topOfRecycledChunk, NULL, entryNext)) {
-			if (entryNext == _firstUnalignedFreeEntry) {
-				_prevFirstUnalignedFreeEntry = (MM_HeapLinkedFreeHeader *)addrTop;
+			if (entryNext == _firstCardUnalignedFreeEntry) {
+				_prevFirstCardUnalignedFreeEntry = (MM_HeapLinkedFreeHeader *)addrTop;
 			}
```

Comment:
now that you touched this code, and noticing that you also use this pattern once, you could introduce a wrapper:
updatePrevCardUnalignedFreeEntry(void * entryNext, void *value) {
		if (entryNext == _firstCardUnalignedFreeEntry) {
				_prevCardUnalignedFreeEntry = value;
			}
}

notice that I also removed 'First' from _prevFirstCardUnalignedFreeEntry (so it's now  _prevCardUnalignedFreeEntry) and from the name of the new helper